### ISSUE TITLE
fromNow builtin behaves differently in different languages

### ISSUE THREAD

**Issue by djmitche** [taskcluster/json-e#343]
```yaml
title:    fromNow with redefined `now`
context:  {}
template: {$let: {now: '2017-01-01T01:00:00.123Z'}, in: {$eval: "fromNow('1 day')"}}
result:   '2017-01-02T01:00:00.123Z'
```

fails on JS and Python and succeeds on Go.  This is similar to #342 and will probably have a similar solution.  @Nicrii if it's easy to fix both at once, that'd be cool, otherwise I'm happy to fix this up following the approach you choose for #342.

**Comment by owlishDeveloper** [comment]
I have a question that is similar to the one I left in #342 - can you tell us more about the practical problem you are trying to solve? I am asking because the `fromNow` built-in accepts 2 arguments: the distance (`str` argument - in this case it's `1 day`) and the optional starting point (`reference` argument) - if you do

```yml
title:    fromNow with redefined `now`
context:  {}
template: {$eval: "fromNow('1 day', '2017-01-01T01:00:00.123Z')"}
result:   '2017-01-02T01:00:00.123Z'
```
as opposed to defining a variable, the test case will pass. So I am curious to learn more about the context (pun not intended)

**Comment by djmitche** [comment]
@Nicrii fixed this in #345
